activerecord/lib/active_record/persistence.rb

```diff
@@ -499,6 +499,16 @@ def query_constraints_list # :nodoc:
         end
       end
 
+      # Always returns an array of column names to be used in querying.
+      # Regardless whether the source is the primary key or the query_constraints configuration.
+      # Supposed to be used internally in Rails in places where the primary key value can always be treated as an array.
+      # This simpifies the code as it avoids branches `on query_constraints_list.nil?`
+      # Long-term this method can be removed and behavior added to the query_constraints_list itself.
+      # However until then we need to keep `query_constraints_list` nillable for backwards compatibility.
```

Comment:
This documentation needs improvement, it's a bit more wordy than we need here. Can you replace it with something like

```
Returns an array of column names to be used in queries. The source of column 
names is derived from +query_constraints_list+ or +primary_key+. This method 
is for internal use when the primary key is to be treated as an array.
```

Reply:
Changed